docs/apps/rbac.md

```diff
@@ -0,0 +1,538 @@
+
+```
+from awx.ansible_base.models.rbac import RoleDefinition
+
+rd = RoleDefinition.objects.get_or_create(name='JT-execute', permissions=['execute_jobtemplate', 'view_jobtemplate'])
+```
+
+### Assigning Permissions
+
+With a role definition object like `rd` above, you can then give out permissions to objects.
+ - `rd.give_permission(user, obj)` - give permissions listed in `rd` to that user for just that object
+ - `rd.remove_permission(user, obj)` - remove permission granted by only that role
+ - `rd.give_global_permission(user)` - if configured, give user execute/view permissions to all job templates in the system
+ - `rd.remove_global_permission(user)`
+
+These cases assume an `obj` of a model type tracked by the RBAC system,
+All the `give_permission` method will return a `RoleUserAssignment` or `RoleTeamAssignment` object.
+The `remove_permission` method will return nothing.
+Assignments have an associated `object_role` in case you need that.
+Removing permission will delete the object role if no other assignments exist.
+
+### Registering Models
+
+Any Django Model (except your user model) can
+be made into a resource in the RBAC system by registering that resource in the registry.
+
+```
+from awx.ansible_base.utils.permission_registry import permission_registry
+
+permission_registry.register(MyModel, parent_field_name='organization')
+```
+
+If you need to manage permissions for special actions beyond the default permissions like "execute", these need to
+be added in the model's `Meta` according to the Django documentation for `auth.Permission`.
+The `parent_field_name` must be a ForeignKey to another model which is the RBAC "parent"
+model of `MyModel`, or `None`.
+
+TODO: Update to allow ManyToMany parent relationship in addition to ForeignKey
+https://github.com/ansible/django-ansible-base/issues/78
+
+#### Django Model, Apps, and Permission Constraints
+
+It is fine to register models from multiple apps, but among the registered models,
+the `._meta.model_name` must be unique. If, in the above example,
+you registered `my_app.mymodel` then it is fine if other models by that same
+name exist like `your_app.mymodel`, but _registering both_ models will throw an error.
+
+A similar constraint exists for the permission `codename`.
+If the `Meta` of `MyModel` lists `permissions = [("move", "Can move my model")]`,
+then no other _registered_ model should list that permission.
+
+This rules out things like registering a model, and also registering a proxy
+model _of that model_.
+
+### Parent Resources
+
+Assuming `obj` has a related `organization` which was declared by `parent_field_name` when registering,
+you can give and remove permissions to all objects of that type "in" the organization
+(meaning they have that ForeignKey to the organization).
+- `rd.give_permission(user, organization)` - give execute/view permissions to all job templates in that organization
```

Comment:
Maybe add a link in here reminding that you defined `rd` as pointing to job templates to guide users as to why this allows all JTs in an organization.